code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp

```diff
@@ -1678,6 +1678,66 @@ static void Init()
 
 		return steeringData ? steeringData->steeringAngle * (180.0f / pi) : 0.0f;
 	}));
+
+	fx::ScriptEngine::RegisterNativeHandler("GET_GAME_POOL", [](fx::ScriptContext& context) {
+		// get the current resource manager
+		auto resourceManager = fx::ResourceManager::GetCurrent();
+
+		// get the owning server instance
+		auto instance = resourceManager->GetComponent<fx::ServerInstanceBaseRef>()->Get();
+
+		// get the server's game state
+		auto gameState = instance->GetComponent<fx::ServerGameState>();
+		std::string pool = context.CheckArgument<const char*>(0);
+
+		std::vector<int> poolList;
+		std::shared_lock<std::shared_mutex> lock(gameState->m_entityListMutex);
+		for (auto& entity : gameState->m_entityList)
+		{
+			switch (entity->type)
+			{
+#if STATE_RDR3
+				case fx::sync::NetObjEntityType::Animal:
+				case fx::sync::NetObjEntityType::Horse:
+#endif
+				case fx::sync::NetObjEntityType::Ped:
+				case fx::sync::NetObjEntityType::Player:
+					if (pool.compare("CPed") == 0)
```

Comment:
Wouldn't it be better to put these outside the switch so you're not constantly doing the compare?